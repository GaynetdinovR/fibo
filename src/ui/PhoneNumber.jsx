import styles from "../styles/Ui.module.sass";
import classNames from "classnames";

const PhoneNumber = ({ phoneNumber, className }) => {
	const formatPhoneNumber = () => {
		return phoneNumber
			.split("")
			.filter(item => !isNaN(item) && item != ' ')
			.join('')
	};

	console.log(formatPhoneNumber(phoneNumber));

	return (
		<a href={`tel:${formatPhoneNumber(phoneNumber)}`} className={classNames(className, styles.phone_number)}>
			{phoneNumber}
		</a>
	);
};

export default PhoneNumber;