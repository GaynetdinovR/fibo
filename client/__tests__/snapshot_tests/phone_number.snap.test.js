import renderer from "react-test-renderer";
import { it, expect } from "@jest/globals";
import PhoneNumber from "../../src/ui/PhoneNumber.jsx";

it("renders correctly", () => {
	const tree = renderer.create(<PhoneNumber phoneNumber={"8 499 391-84-49"}/>).toJSON();
	expect(tree).toMatchSnapshot();
});
