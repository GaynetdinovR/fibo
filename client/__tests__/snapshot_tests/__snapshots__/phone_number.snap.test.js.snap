// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="phone_number"
  href="tel:84993918449"
>
  8 499 391-84-49
</a>
`;
